core/src/main/java/org/renjin/primitives/S3.java

```diff
@@ -259,6 +261,209 @@ public static SEXP tryDispatchFromPrimitive(Context context, Environment rho, Fu
     return method.doApply(context, rho, newArgs);
   }
 
+  private static SEXP handleS4object(@Current Context context, SEXP source, String functionName, PairList args,
+                                     Environment rho, int argumentIndex, ArrayList<String> allArgClasses) {
+    // for now the namespace is hardcoded (":base") since all primitives are from the base package. But to
+    // allow this function to also work with standardGeneric the namespace needs to be identified at runtime
+    // since these standardGenerics migh be in .GlobalEnv or elsewhere.
+    String functionEnvName = ".__T__" + functionName + ":base";
```

Comment:
I see this string concatentation in several places. Would be preferable to extract this out to separate methods or a whole class that provides the metadata keys.